src/phase_change/CarmanKozenyDragForce.cpp

```diff
@@ -0,0 +1,372 @@
+                double penalty_rho_scale = 0.0, penalty_mu_scale = 0.0;
+                if (d_use_rho_scale)
+                {
+                    penalty_rho_scale = (*rho_data)(s_i) / dt;
+                }
+
+                if (d_use_mu_scale)
+                {
+                    const double mu_lower = (*mu_data)(s_i.toCell(0));
+                    const double mu_upper = (*mu_data)(s_i.toCell(1));
+                    const double mu = 0.5 * (mu_lower + mu_upper);
+                    penalty_mu_scale = mu / (h_min * h_min);
+                }
+                const double penalty = d_penalty_factor * (penalty_rho_scale + penalty_mu_scale);
+                (*u_data)(s_i) = penalty * alpha_s * alpha_s / (std::pow(1.0 - alpha_s, 3.0) + d_ed);
+            }
+        }
+    }
+
+    return;
+} // demarcateBrinkmanZone
+
+void
+CarmanKozenyDragForce::postprocessComputeBrinkmanPenalization(double current_time, double new_time, int num_cycles)
+{
+    BrinkmanPenalizationStrategy::postprocessComputeBrinkmanPenalization(current_time, new_time, num_cycles);
+
+    return;
+} // postprocessComputeBrinkmanPenalization
+
+void
+CarmanKozenyDragForce::putToDatabase(Pointer<Database> db)
+{
+    db->putDouble("d_penalty_factor", d_penalty_factor);
+    db->putBool("d_use_rho_scale", d_use_rho_scale);
+    db->putBool("d_use_mu_scale", d_use_mu_scale);
+    db->putDouble("d_ed", d_ed);
+    return;
+} // putToDatabase
+
+/////////////////////////////// PRIVATE //////////////////////////////////////
+
+void
+CarmanKozenyDragForce::getFromInput(Pointer<Database> input_db, bool is_from_restart)
+{
+    if (!is_from_restart)
+    {
+        if (input_db->keyExists("penalty_factor"))
+        {
+            d_penalty_factor = input_db->getDouble("penalty_factor");
+        }
+
+        if (input_db->keyExists("use_rho_scale"))
+        {
+            d_use_rho_scale = input_db->getBool("use_rho_scale");
+        }
+
+        if (input_db->keyExists("use_mu_scale"))
+        {
+            d_use_mu_scale = input_db->getBool("use_mu_scale");
+        }
+
+        if (input_db->keyExists("avoid_division_by_zero_factor"))
+        {
+            d_ed = input_db->getDouble("avoid_division_by_zero_factor");
+        }
+    }
+    return;
+} // getFromInput
+
+void
+CarmanKozenyDragForce::getFromRestart()
+{
+    Pointer<Database> restart_db = RestartManager::getManager()->getRootDatabase();
+    Pointer<Database> db;
+    if (restart_db->isDatabase(d_object_name))
+    {
+        db = restart_db->getDatabase(d_object_name);
+    }
+    else
+    {
+        TBOX_ERROR(d_object_name << ":  Restart database corresponding to " << d_object_name
+                                 << " not found in restart file." << std::endl);
+    }
+    d_penalty_factor = db->getDouble("d_penalty_factor");
+    d_use_rho_scale = db->getBool("d_use_rho_scale");
+    d_use_mu_scale = db->getBool("d_use_mu_scale");
+    d_ed = db->getDouble("d_ed");
```

Comment:
```suggestion
    avoid_division_by_zero_factor = db->getDouble("avoid_division_by_zero_factor");
```

Here and elsewhere - we should use the same names for everything in the input database/class/restart database (with the exception of the `d_` prefix)